### PR TITLE
mock support for Navy Linux build

### PR DESCRIPTION
**Pull request by unixlabs**
Hi,

 code commit for the Navy Linux mock configuration file. 

Thanks.

### CODE REVIEW

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 1]
```diff
@@ -0,0 +1,69 @@
+config_opts['chroot_setup_cmd'] = 'install tar gcc-c++ redhat-rpm-config system-release which xz sed make bzip2 gzip gcc coreutils unzip shadow-utils diffutils cpio bash gawk rpm-build info patch util-linux findutils grep'
+config_opts['dist'] = 'el8'  # only useful for --resultdir variable subst
+config_opts['releasever'] = '8'
+config_opts['package_manager'] = 'yum'
+config_opts['extra_chroot_dirs'] = [ '/run/lock', ]
+config_opts['macros']['%dist'] = '.el8'
+config_opts['bootstrap_image'] = 'navylinux:8'
+config_opts['macros']['vendor'] = "The Navy Linux Project."
```
Comment (xsuchy) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:8]: Really the trailing dot?

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+config_opts['macros']['packager'] = "packager@navylinux.org"
```
Comment (xsuchy) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:9]: The macros vendor and packager are usually set by the build system. Do you really want to set them here?

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+protected_packages=
+module_platform_id=platform:el8
+user_agent={{ user_agent }}
+
+[nl-base]
+name=Navy Linux Enterprise OS Repository - el$releasever
+baseurl=https://cdn.navylinux.org/navylinux/releases/8.4r1/x86_64/os/
+enabled=1
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-navy-linux-official
```
Comment (xsuchy) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:36]: This file does not exist on my workstation, so the build will fail. There are several ways how to handle it. The easiest one is to add the key to https://github.com/xsuchy/distribution-gpg-keys/ and use the key from that package (every other config does that).

**Review by xsuchy — CHANGES_REQUESTED** [review]

File: mock-core-configs/etc/mock/templates/navy-8.tpl (hunk 3 above)
Comment (unixlabs) [reply mock-core-configs/etc/mock/templates/navy-8.tpl]: Ok, I will follow up thanks for the help.

File: mock-core-configs/etc/mock/templates/navy-8.tpl (hunk 2 above)
Comment (unixlabs) [reply mock-core-configs/etc/mock/templates/navy-8.tpl]: fixed. Thanks for sharing

**Comment by praiskup** [comment]
Please take a look at the [new docs section](https://rpm-software-management.github.io/mock/feature-gpg-and-ssl)
IOW, what Mirek wanted above is to add the pubkey into the [distribution-gpg-keys project](https://github.com/xsuchy/distribution-gpg-keys).

**Comment by praiskup** [comment]
Also, we will need to bump the `Requires: distribution-gpg-keys` bumped to 1.60+

**Comment by praiskup** [comment]
@unixlabs can you bump [the requirement](https://github.com/rpm-software-management/mock/blob/2dcb7ce02d5d13ce6ad5fa9712423f2038e6dc29/mock-core-configs/mock-core-configs.spec#L20-L21) please?

**Comment by praiskup** [comment]
Thanks for the update!

I tested the config, and I got this error:
```
Failed loading plugin "upload-profile": module 'collections' has no attribute 'MutableMapping'
Navy Linux Enterprise OS Repository - el8                                                                                                                                                                                                                                                                             411  B/s | 153  B     00:00    
Errors during downloading metadata for repository 'nl-base':
  - Status code: 404 for https://cdn.navylinux.org/navylinux/releases/$releaseversion/x86_64/os/repodata/repomd.xml (IP: 5.196.144.116)
Error: Failed to download metadata for repo 'nl-base': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
```
Looks like a typo

**Comment by praiskup** [comment]
Looks OK, I'm going to merge.

**Comment by praiskup** [comment]
Thank you!

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 4]
```diff
@@ -0,0 +1,91 @@
+config_opts['chroot_setup_cmd'] = 'install tar gcc-c++ redhat-rpm-config system-release which xz sed make bzip2 gzip gcc coreutils unzip shadow-utils diffutils cpio bash gawk rpm-build info patch util-linux findutils grep'
+config_opts['dist'] = 'el8'  # only useful for --resultdir variable subst
+config_opts['releasever'] = '8'
+config_opts['package_manager'] = 'yum'
```
Comment (praiskup) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:4]: Not yet ... are you sure `yum` is the correct value?

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+config_opts['extra_chroot_dirs'] = [ '/run/lock', ]
+config_opts['releaseversion'] = '8.4r1'
+
+config_opts['dnf.conf'] = """
+[main]
+keepcache=1
+debuglevel=2
+reposdir=/dev/null
+logfile=/var/log/yum.log
```
Comment (praiskup) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:13]: The same here...

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+gpgcheck=1
+gpgkey=file:///usr/share/distribution-gpg-keys/navy/RPM-GPG-KEY-navy-linux-official
+
+[epel]
+name=Extra Packages for Enterprise Linux $releasever 
+baseurl=https://dl.fedoraproject.org/pub/epel/$releasever/Everything/x86_64/
+failovermethod=priority
+gpgkey=file:///usr/share/distribution-gpg-keys/epel/RPM-GPG-KEY-EPEL-8
+gpgcheck=1
+skip_if_unavailable=False
```
Comment (Conan-Kudo) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:87]: This should not be in here, we have EPEL configuration already.

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+retries=20
+obsoletes=1
+gpgcheck=0
+assumeyes=1
+syslog_ident=mock
+syslog_device=
+mdpolicy=group:primary
+best=1
+protected_packages=
+module_platform_id=platform:el$releasever
```
Comment (Conan-Kudo) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:22]: This isn't actually valid, you need to use either `8` or `{{ releasever }}`

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: mock-core-configs/etc/mock/templates/navy-8.tpl (hunk 7 above)
Comment (unixlabs) [reply mock-core-configs/etc/mock/templates/navy-8.tpl]: thanks for the correction.

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+syslog_device=
+mdpolicy=group:primary
+best=1
+protected_packages=
+module_platform_id=platform:el8
+user_agent={{ user_agent }}
+
+[nl-base]
+name=Navy Linux Enterprise OS Repository - el$releasever
+baseurl=https://cdn.navylinux.org/navylinux/releases/$releasever/x86_64/os/
```
Comment (Conan-Kudo) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:27]: `x86_64` -> `$basearch`

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+enabled=1
+gpgcheck=1
+gpgkey=file:///usr/share/distribution-gpg-keys/navy/RPM-GPG-KEY-navy-linux-official
+
+[nl-every]
+name=Navy Linux Enterprise Every Repository - el$releasever
+baseurl=https://cdn.navylinux.org/navylinux/releases/$releasever/x86_64/everything/
```
Comment (Conan-Kudo) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:34]: `x86_64` -> `$basearch`

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+enabled=1
+gpgcheck=1
+gpgkey=file:///usr/share/distribution-gpg-keys/navy/RPM-GPG-KEY-navy-linux-official
+
+[nl-powertools]
+name=Navy Linux Enterprise Power Repository - el$releasever
+baseurl=https://cdn.navylinux.org/navylinux/releases/$releasever/x86_64/powertools/
```
Comment (Conan-Kudo) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:41]: `x86_64` -> `$basearch`

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+enabled=1
+gpgcheck=1
+gpgkey=file:///usr/share/distribution-gpg-keys/navy/RPM-GPG-KEY-navy-linux-official
+
+[nl-kernel]
+name=Navy Linux Enterprise Kernel  Repository - el$releasever
+baseurl=https://cdn.navylinux.org/navylinux/releases/$releasever/x86_64/kernel/
```
Comment (Conan-Kudo) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:48]: `x86_64` -> `$basearch`

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+enabled=1
+gpgcheck=1
+gpgkey=file:///usr/share/distribution-gpg-keys/navy/RPM-GPG-KEY-navy-linux-official
+
+[nl-extra]
+name=Navy Linux Enterprise Extra  Repository - el$releasever
+baseurl=https://cdn.navylinux.org/navylinux/releases/$releasever/x86_64/extra/
```
Comment (Conan-Kudo) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:55]: `x86_64` -> `$basearch`

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+enabled=0
+gpgcheck=1
+gpgkey=file:///usr/share/distribution-gpg-keys/navy/RPM-GPG-KEY-navy-linux-official
+
+[nl-devel]
+name=Navy Linux Devel Repository - el$releasever
+baseurl=https://cdn.navylinux.org/navylinux/releases/$releasever/x86_64/devel/
```
Comment (Conan-Kudo) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:62]: `x86_64` -> `$basearch`

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+enabled=0
+gpgcheck=1
+gpgkey=file:///usr/share/distribution-gpg-keys/navy/RPM-GPG-KEY-navy-linux-official
+
+[nl-debug]
+name=Navy Linux Debug Repository - el$releasever
+baseurl=https://cdn.navylinux.org/navylinux/releases/$releasever/x86_64/debug/
```
Comment (Conan-Kudo) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:69]: `x86_64` -> `$basearch`

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+enabled=0
+gpgcheck=1
+gpgkey=file:///usr/share/distribution-gpg-keys/navy/RPM-GPG-KEY-navy-linux-official
+
+[nl-source]
+name=Navy Linux Source Repository - el$releasever
+baseurl=https://cdn.navylinux.org/navylinux/releases/$releasever/x86_64/source/
```
Comment (Conan-Kudo) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:76]: `x86_64` -> `$basearch`

File: mock-core-configs/etc/mock/templates/navy-8.tpl [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+
+[nl-base]
+name=Navy Linux Enterprise OS Repository - el$releasever
+baseurl=https://cdn.navylinux.org/navylinux/releases/$releasever/x86_64/os/
+enabled=1
+gpgcheck=1
+gpgkey=file:///usr/share/distribution-gpg-keys/navy/RPM-GPG-KEY-navy-linux-official
+
+[nl-every]
+name=Navy Linux Enterprise Every Repository - el$releasever
```
Comment (Conan-Kudo) [inline mock-core-configs/etc/mock/templates/navy-8.tpl:33]: Shouldn't this say `Everything Repository`? That's what it is, after all...

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: mock-core-configs/etc/mock/templates/navy-8.tpl (hunk 16 above)
Comment (unixlabs) [reply mock-core-configs/etc/mock/templates/navy-8.tpl:33]: nl-every is an Appsteam in Red Hat

File: mock-core-configs/etc/mock/navy-8-x86_64.cfg [hunk 17]
```diff
@@ -0,0 +1,4 @@
+config_opts['root'] = 'navy-8-x86_64'
```
Comment (praiskup) [inline mock-core-configs/etc/mock/navy-8-x86_64.cfg:1]: This also usually goes to the template, but it's a nit ...

**Comment by praiskup** [comment]
Thanks, merged.  Feel free to propose follow-up patches.

**Pull request closed without merging**